app/src/main/java/com/example/cheatchess/ChessModel.kt

```diff
@@ -85,10 +88,57 @@ class ChessModel {
     }
 
     private fun isOnBoard(chessPiece: ChessPiece) : Boolean {
-        return (0..7).contains(chessPiece.row)
+        return (BOARD_RANGE).contains(chessPiece.row)
     }
 
     fun changeTurn() {
         isWhiteTurn = !isWhiteTurn
     }
+
+    fun positionToFEN() : String {
```

Comment:
What is FEN?

Reply:
FEN (Forsyth–Edwards Notation) is a chess notation. It's an intuitive way you can write and read an entire chess position from a short string.

Reply:
The wiki page explains it really well and quick if you're interested.